pkg/cli/cmd/radinit/recipe.go

```diff
@@ -52,112 +70,88 @@ func NewDevRecipeClient() DevRecipeClient {
 // GetDevRecipes is a function that queries a registry for recipes with a specific tag and returns a map of recipes.
 // If an error occurs, an error is returned.
 func (drc *devRecipeClient) GetDevRecipes(ctx context.Context) (map[string]map[string]corerp.RecipePropertiesClassification, error) {
-	reg, err := remote.NewRegistry(DevRecipesRegistry)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create client to registry %s -  %s", DevRecipesRegistry, err.Error())
-	}
+	logger := ucplog.FromContextOrDiscard(ctx)
 
 	// The tag will be the major.minor version of the release.
 	tag := version.Channel()
 	if version.IsEdgeChannel() {
 		tag = "latest"
 	}
 
-	// Temporary solution to get all repositories.
-	// The issue is that if RepositoryListPageSize is not specified the default is 100.
-	// We have 104 repositories in the registry as of 12 Oct 2023. That is why processRepositories
-	// function was being called twice and the second call was overwriting all the recipes.
-	// TODO: Remove this once we have a better solution.
-	reg.RepositoryListPageSize = 1000
-
-	recipes := map[string]map[string]corerp.RecipePropertiesClassification{}
-
-	// if repository has the correct path it should look like: <registryPath>/recipes/<category>/<type>:<tag>
-	// Ex: ghcr.io/radius-project/recipes/local-dev/rediscaches:0.20
-	// The start parameter is set to "radius-rp" because our recipes are after that repository.
-	err = reg.Repositories(ctx, "", func(repos []string) error {
-		// validRepos will contain the repositories that have the requested tag.
-		validRepos := []string{}
-		for _, repo := range repos {
-			r, err := reg.Repository(ctx, repo)
-			if err != nil {
-				continue
-			}
+	validRepos := []string{}
```

Comment:
So valid repos are only the ones that start with the recipe prefix, correct?

Reply:
I simplified the logic a bit more.